### PR TITLE
search UI additions

### PR DESCRIPTION
**Pull request by djvoa12**
**Summary:**
- Shows course and session tags
- Initially show up to three session (show more button to show the rest)
- Added frontend pagination

**UI:**
search

Fixes https://github.com/ilios/frontend/issues/4608
Fixes https://github.com/ilios/frontend/issues/4610
Fixes https://github.com/ilios/frontend/issues/4611

### CODE REVIEW

File: app/components/course-search-result.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+export default Component.extend({
+  classNames: ['course-search-result', 'results-row'],
+  tagName: 'li',
+
+  course: null,
+  showMore: false,
+
+  courseTags: reads('course.matchedIn'),
+  sessions: reads('course.sessions'),
```
Comment (jrjohnson) [inline app/components/course-search-result.js]: Remove this property and just call the `computedSessions` property `sessions`.

File: app/components/pagination-links.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  disableNext: computed('lastPage', 'page', function() {
+    return this.lastPage === this.page;
+  }),
+
+  lastPage: computed('results', 'size', function() {
+    return Math.ceil(this.results.length / this.size);
+  }),
+
+  computedPages: computed('page', 'totalPages', function() {
```
Comment (jrjohnson) [inline app/components/pagination-links.js]: Just `pages` please. calling it `computedPages` is redundant.

File: app/controllers/search.js [hunk 3]
```diff
@@ -1,9 +1,18 @@
 import Controller from '@ember/controller';
 
 export default Controller.extend({
-  queryParams: {
-    query: 'q'
-  },
+  queryParams: ['page', { query: 'q' }],
```
Comment (jrjohnson) [inline app/controllers/search.js]: This is much more confusing than
```
queryParams: {
    query: 'q',
    page: 'page'
  },
```

File: app/controllers/search.js [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
 
-  query: ''
+  page: 1,
+  query: '',
+
+  actions: {
+    prevPage() {
+      this.set('page', this.page - 1);
```
Comment (jrjohnson) [inline app/controllers/search.js]: Can you click on a page number directly? In that case it seems like both of these actions should just be replaced with `setPage` and the computation can be done in the component itself.

File: app/styles/components/course-search-result.scss [hunk 5]
```diff
@@ -0,0 +1,31 @@
+.course-search-result {
+  margin-bottom: 1rem;
+  max-width: 40rem;
+
+  .matched-in {
```
Comment (jrjohnson) [inline app/styles/components/course-search-result.scss]: is `matched-in` different from `global-search-tags`?

File: app/styles/components/course-search-result.scss [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+  .global-search-tags {
+    margin: .125rem 0;
+    overflow: hidden;
+
+    .global-search-tag {
+      background: lighten($grey, 5%);
+      border: 1px solid lighten($dark-grey, 10%);
+      border-radius: .125rem;
+      color: $darker-grey;
```
Comment (jrjohnson) [inline app/styles/components/course-search-result.scss]: There isn't enough contrast here to be legible. The background needs to be lighter or else the color should be darker.

File: app/styles/components/pagination-links.scss [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    outline: none;
+    padding: .125rem 0;
+
+    &:hover {
+      background: $ilios-orange;
+      border: 1px solid $ilios-orange;
+      color: $white;
+    }
+
+    @media only screen and (max-width: 550px) {
```
Comment (jrjohnson) [inline app/styles/components/pagination-links.scss]: This looks amazing! Love how it collapses down. Really nice.

File: app/templates/components/global-search-tags.hbs [hunk 8]
```diff
@@ -0,0 +1,17 @@
+{{#if @tags}}
+  <div class="global-search-tags">
+    {{#each @tags as |tag|}}
+      <span class="global-search-tag">
+        {{#if (eq tag "id")}}
```
Comment (jrjohnson) [inline app/templates/components/global-search-tags.hbs]: isn't this the same as the last `concat`ed option?

File: app/components/course-search-result.js (hunk 1 above)
Comment (djvoa12) [reply app/components/course-search-result.js]: 👍

File: app/components/pagination-links.js (hunk 2 above)
Comment (djvoa12) [reply app/components/pagination-links.js]: 👍

File: app/controllers/search.js (hunk 3 above)
Comment (djvoa12) [reply app/controllers/search.js]: 👍

File: app/controllers/search.js (hunk 4 above)
Comment (djvoa12) [reply app/controllers/search.js]: 👍

File: app/styles/components/course-search-result.scss (hunk 5 above)
Comment (djvoa12) [reply app/styles/components/course-search-result.scss]: This needs to be removed.

File: app/templates/components/global-search-tags.hbs (hunk 8 above)
Comment (djvoa12) [reply app/templates/components/global-search-tags.hbs]: Yes.

File: app/styles/components/course-search-result.scss (hunk 6 above)
Comment (djvoa12) [reply app/styles/components/course-search-result.scss]: I will lighten the background a little more.

File: app/components/course-search-result.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { reads } from '@ember/object/computed';
+
+export default Component.extend({
+  classNames: ['course-search-result', 'results-row'],
```
Comment (jrjohnson) [inline app/components/course-search-result.js]: Is results-row still used? Looks like it's gone form the css.

File: app/styles/components/course-search-result.scss (hunk 6 above)
Comment (jrjohnson) [reply app/styles/components/course-search-result.scss]: Still not great, but I'm ok with it for now.

File: app/components/course-search-result.js (hunk 9 above)
Comment (djvoa12) [reply app/components/course-search-result.js]: Will remove.

**Pull request merged**